model/task/task.go

```diff
@@ -1050,61 +1268,126 @@ func (t *Task) UpdateHeartbeat() error {
 	)
 }
 
-// SetPriority sets the priority of the tasks and the tasks that they depend on
+// SetPriority sets the priority of the task.
+// It increases the priority of the tasks it depends on.
 func (t *Task) SetPriority(priority int64, user string) error {
 	t.Priority = priority
-	modifier := bson.M{PriorityKey: priority}
-
-	//blacklisted - this task should never run, so unschedule it now
-	if priority < 0 {
-		modifier[ActivatedKey] = false
-	}
-
-	ids, err := t.getRecursiveDependencies()
+	depTasks, err := GetRecursiveDependenciesUp([]Task{*t}, nil)
 	if err != nil {
 		return errors.Wrap(err, "error getting task dependencies")
 	}
+
+	ids := []string{t.Id}
 	ids = append(ids, t.ExecutionTasks...)
+	depIDs := make([]string, 0, len(depTasks))
+	for _, task := range depTasks {
+		depIDs = append(depIDs, task.Id)
+	}
 
 	_, err = UpdateAll(
 		bson.M{"$or": []bson.M{
-			{IdKey: t.Id},
-			{IdKey: bson.M{"$in": ids},
-				PriorityKey: bson.M{"$lt": priority}},
+			{IdKey: bson.M{"$in": ids}},
+			{
+				IdKey:       bson.M{"$in": depIDs},
+				PriorityKey: bson.M{"$lt": priority},
+			},
 		}},
-		bson.M{"$set": modifier},
+		bson.M{"$set": bson.M{PriorityKey: priority}},
 	)
+	if err != nil {
+		return errors.Wrap(err, "can't update priority")
+	}
 
 	event.LogTaskPriority(t.Id, t.Execution, user, priority)
```

Comment:
Do we need to also log for the dependencies?

Reply:
good point!
Done